### PR TITLE
Enable CK Attention for Navi31

### PR DESCRIPTION
**Pull request by hyoon1**
- Enables CK Attention for Navi31
- Requires this branch of Flash Attention:
- https://github.com/ROCm/flash-attention/tree/howiejay/navi_support

### CODE REVIEW

File: vllm/attention/backends/rocm_flash_attn.py [hunk 1]
```diff
@@ -509,16 +518,18 @@ def __init__(
...
+            # if not using triton, navi21/navi10 do not use flash-attn
             # either
             if not current_platform.has_device_capability(90):
                 self.use_naive_attn = True
             else:
-                try:
-                    from flash_attn import flash_attn_varlen_func  # noqa: F401
-                    self.attn_func = flash_attn_varlen_func
+                if flash_attn_available:
+                    if current_platform.has_device_capability(110):
```
Comment (gshtras) [inline vllm/attention/backends/rocm_flash_attn.py:527]: This check isn't equivalent to is_navi
On Cuda device_capability is meant to increase with each new architecture, and is meant to differentiate by new features support, such as FP8, etc.
On ROCm it is 1st digit of gfx * 10 + 2nd digit, which doesn't mean much, especially for any future architectures.

**Review by gshtras — CHANGES_REQUESTED** [review]
Is this similar to https://github.com/ROCm/vllm/pull/281
Can the 2 efforts be combined?

Please also try to simplify the conditions, with this level of nested elseifs it's hard to follow the logic.

**Comment by hyoon1** [comment]
> Is this similar to #281 Can the 2 efforts be combined?
> 
> Please also try to simplify the conditions, with this level of nested elseifs it's hard to follow the logic.

PR #281 is specific to a particular vision model and does not call the path used in general LLM models. Therefore, it seems difficult to merge. The condition has been simplified to match the depth in the existing code.

File: vllm/attention/backends/rocm_flash_attn.py (hunk 1 above)
Comment (hyoon1) [reply vllm/attention/backends/rocm_flash_attn.py]: updated
